lib/apple_sign_in/token.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module AppleSignIn
+  class Token
+    APPLE_AUD = 'https://appleid.apple.com'
+    APPLE_TOKEN_URL = 'https://appleid.apple.com/auth/token'
+
+    attr_reader :grant_type, :code, :refresh_token
+
+    def initialize(type, code, refresh_token)
+      @grant_type = type
+      @code = code
+      @refresh_token = refresh_token
+    end
+
+    def authenticate
+      uri = URI.parse(APPLE_TOKEN_URL)
+      http = Net::HTTP.new(uri.host, uri.port)
+      request = Net::HTTP::Post.new(uri.request_uri, request_header)
+      request.body = apple_token_params.to_json
+      http.request(request)
+    end
+
+
+    private
+
+    def apple_token_params
+      {
+        client_id: AppleSignIn.config.apple_team_id,
+        client_secret: client_secret_from_jwt,
+        code: code,
+        grant_type: grant_type,
+        refresh_token: refresh_token,
+        redirect_uri: AppleSignIn.config.redirect_uri
+      }
+    end
+
+    def client_secret_from_jwt
+      JWT.encode(claims, AppleSignIn.config.apple_private_key, 'ES256', cliams_headers)
+    end
+
+    def claims
+      {
+        iss: AppleSignIn.config.apple_team_id,
+        iat: Time.now.to_i,
+        exp: Time.now.to_i + 10.minutes.to_i,
```

Comment:
just curious, 10 minutes is because of any documentation or because we decided that?

Reply:
I don't have a reason, i only chose a time value, because i need to test this things in a real case to know how it works. Probably, this value could be a configuration of the initializer but i need to be sure.

Reply:
no worries, just curious where this came from, im completely fine with the actual reason

Reply:
For what I saw,  10 minutes is correct